src/core/components/LoginPage/index.js

```diff
@@ -20,7 +21,7 @@ export default class LoginPage extends React.Component {
           {message || _('You must be logged in to access this page.')}
         </p>
         <p>
-          <a className="button" href={startLoginUrl()}>
+          <a className="button" href={startLoginUrl({ location })}>
```

Comment:
This looks pretty similar what `<footer>{this.logIn()}</footer>` does except it doesn't handle the already logged in case. Is this something you're going to fix up later? (I'd be fine with that.)